httpclient/api_client.go

```diff
@@ -171,8 +171,15 @@ func (c *ApiClient) isRetriable(ctx context.Context, err error) bool {
 	if c.config.ErrorRetriable(ctx, err) {
 		return true
 	}
+	var skipRetryOnIO bool
+	// depend on the HTTP fixture interface to prevent any coupling
+	if skippable, ok := c.httpClient.Transport.(interface {
+		SkipRetryOnIO() bool
+	}); ok {
+		skipRetryOnIO = skippable.SkipRetryOnIO()
+	}
 	_, isIO := err.(*url.Error)
-	if isIO {
+	if isIO && !skipRetryOnIO {
```

Comment:
Where is this used/needed? I didn't see a reference to it in the linked file.

Reply:
if we don't do this - we keep retrying for the default 5 minutes.